### DISCUSSION TITLE
EMQX Docker Cluster Installation with Load Balancer or DNS Configuration

### DISCUSSION THREAD

**Discussion by pitambariu** [emqx/emqx#14435] [Q&A] [answered]
Hi Team,

Could someone please help me set up an EMQX 3-node cluster across three different nodes using AWS Load Balancer and Auto Scaling?

I have attempted the configuration, but the nodes are not connecting to each other.

Thanks in advance!

**Comment by id** [comment]
Hi @pitambariu.

You can take a look at this example with Terraform https://github.com/emqx/emqx-usage-example/tree/main/iov-s3-aws. It's a little more complex than what are you asking about, but does the job.

The trick there is to use static hostnames used in `static_seeds`, and have instances themselves update Route53 records when they come up. The difference with your requirements is that there we are using 1 ASG per AZ, and each ASG is static, min = max = desired = 1, so we can assign one hostname per ASG.

In your scenario you may use similar approach, but use `DNS` discovery and update entries in a single `SRV` record instead https://docs.emqx.com/en/emqx/latest/deploy/cluster/create-cluster.html#configure-autocluster-by-dns-records. Then may be run a Lambda periodically to clean up defunct records.

**Comment by pitambariu** [comment]
version: '3.8'

services:
  emqx:
    image: emqx:latest
    container_name: emqx-node2
    environment:
      - "EMQX_NODE_NAME=emqx@emqx1-poc.xxx.com"
      - node.cookie=secretcookie
      - cluster.discovery_strategy=static
      - "EMQX_CLUSTER_DISCOVERY_STRATEGY=static"
      - "EMQX_CLUSTER_STATIC_SEEDS=emqx@emqx-poc.xxx.com,emqx@emqx1-poc.xxx.com"
      - node.dist_listen_min=4370
      - node.dist_listen_max=4400
      - log.level=debug
      - EMQX_LISTENER__TCP__EXTERNAL__ACCEPTORS=64
      - EMQX_LISTENER__TCP__EXTERNAL__MAX_CLIENTS=1000
      - WITH_EPMD=1
    network_mode: "host"
    ports:
      - "1883:1883"
      - "8883:8883"
      - "18083:18083"
      - "4369:4369"
      - "4370-4400:4370-4400"
    healthcheck:
      test: ["CMD", "emqx_ctl", "status"]
      interval: 30s
      timeout: 10s
      retries: 3
    restart: always

-------
Is this compose file correct

After running the above compose file, the node is unable to connect to the cluster

Please suggest any modifications to the Compose file or any other recommendations, as I am running it on an Ubuntu server

**Reply by id** [reply]
What do `emqx1-poc.xxx.com` and `emqx1-poc.xxx.com` resolve to from EC2 instances?

**Comment by pitambariu** [comment]
Hi
these dns resolved to respective instance IP

Best Regards,


Pitambar Sen *|* DevOps Engineer

M : +91 6360371510 *|* E : ***@***.***
***@***.***>

*-------------------------------------------------------**-**-*

*[image: Infinite Uptime] <https://www.infinite-uptime.com/>*
*www.infinite-uptime.com <http://www.infinite-uptime.com/> **|** Socialize
with us [image: Follow our LinkedIn Page]
<https://www.linkedin.com/company/infinite-uptime/>  [image: Engage with us
on Facebook] <https://www.facebook.com/infiniteuptime/>  [image: Connect on
YouTube] <https://www.youtube.com/channel/UCceOqZfxozvbEm3SDGPS6bg>*

*-------------------------------------------------------**-**-*

[image: Accelerate Industrial Digital Transformation]
<https://www.infinite-uptime.com/digital-reliability>

**Comment by id** [comment]
```
environment:
- EMQX_NODE_NAME="emqx@emqx1-poc.xxx.com"
- EMQX_CLUSTER__DISCOVERY_STRATEGY=static
- EMQX_CLUSTER__STATIC__SEEDS="emqx@emqx-poc.xxx.com,emqx@emqx1-poc.xxx.com"
- EMQX_NODE__DIST_LISTEN_MIN=4370
- EMQX_NODE__DIST_LISTEN_MAX=4400
- EMQX_LOG__CONSOLE_HANDLER__LEVEL=debug
- EMQX_LISTENER__TCP__EXTERNAL__ACCEPTORS=64
- EMQX_LISTENER__TCP__EXTERNAL__MAX_CLIENTS=1000
```

**Reply by pitambariu** [reply]
getting below error 
root@ip-10-1-3-40:/home/ubuntu# docker logs -f emqx-node2
ERROR: Invalid node name, should be of format '^[0-9A-Za-z_-]+$'.
ERROR: Invalid node name, should be of format '^[0-9A-Za-z_-]+$'.

**Reply by id** [reply, marked as answer]
I do not have time to test this right now. Try to put double quotes around the whole line like in this example in the documentation https://docs.emqx.com/en/emqx/latest/deploy/install-docker-ce.html

```
environment:
- "EMQX_NODE_NAME=emqx@emqx1-poc.xxx.com"
- "EMQX_CLUSTER__DISCOVERY_STRATEGY=static"
- "EMQX_CLUSTER__STATIC__SEEDS=[emqx@emqx-poc.xxx.com,emqx@emqx1-poc.xxx.com]"
```

**Reply by pitambariu** [reply]
Thanks it working
